scout/parse/clinvar.py

```diff
@@ -0,0 +1,216 @@
+import urllib.request
+import sys
+from scout.constants import CLIVAR_HEADER, CLINVAR_OPTIONAL, CASEDATA_HEADER, CASEDATA_OPTIONAL
+
+def get_variant_lines(form_fields):
+    """Parses the form from blueprints/variants/clinvar.html
+    and creates the lines for the .Variant.csv clinvar submission file.
+    """
+
+    clinvar_header  = list(CLIVAR_HEADER)
+    optional_fields = dict(CLINVAR_OPTIONAL)
+    clinvars_dict = {}
+
+    if 'all_vars' in form_fields:
+        clinvars = form_fields['local'] #this is a list
+    else:
+        clinvars = [form_fields['main_var'][0]] #also a list, but has one element
+
+    for clinvar in clinvars:
```

Comment:
If there is any possibility to simplify this function it would be nice. Hard to follow at the moment. Perhaps break it into smaller parts

Reply:
I know, it looks quite complicated, but trust me, I re-wrote it for this reason and this is the simplest I could get. The fact is that I'm parsing this huge clinvar submission form and to do it in an elegant way I gave the form fields the same names as the clinvar excel sheet columns. These strings are defined in the constants file.
And I need all these loops to understand which columns I have to include in the generated files (both variants file and case data file). For instance if I have to include 4 variants, some with optional data, some no. Then I have to check if any of the variants has a field filled in and in that case I have to include the field in the document header and an empty cell for the variants where the field is blank instead. On the other hand, if the field is not used I have to remove it from the header so I don't print the column at all.

I think this explanation might be more complicated than the code itself! :)

Anyway, I've tried to document these 2 subroutines a bit better. We can discuss about it!